### PR TITLE
Add rector

### PR DESCRIPTION
**Pull request by xepozz**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️

### CODE REVIEW

File: .github/workflows/rector.yml [hunk 1]
```diff
@@ -0,0 +1,65 @@
+on:
+  push:
```
Comment (arogachev) [inline .github/workflows/rector.yml:2]: Should it be used for pull requests too, like other workflows https://github.com/yiisoft/package-template/blob/852e46eebbf89dd157404a5b27b1d070e79fb194/.github/workflows/build.yml#L2, or it's unnecessary in this case?

**Review by arogachev — APPROVED** [review]

**Comment by arogachev** [comment]
Decided to wait for other approach suggested by @terabytesoftw (using https://github.com/yiisoft/actions/).

File: .github/workflows/rector.yml (hunk 1 above)
Comment (xepozz) [reply .github/workflows/rector.yml:2]: The reason is not call rector twice to prevent duplicating changes

Comment (arogachev) [reply .github/workflows/rector.yml]: OK, got it.

File: README.md [hunk 2]
```diff
@@ -58,6 +58,15 @@ The code is statically analyzed with [Psalm](https://psalm.dev/). To run static
 ./vendor/bin/psalm
 ```
 
+### Rector
+
+Use [Rector](https://github.com/rectorphp/rector) to make codebase follow some specific rules or 
+use either newest or any specific version of PHP API:
```
Comment (arogachev) [inline README.md:64]:
```suggestion
use either newest or any specific version of PHP: 
```

**Review by arogachev — APPROVED** [review]

**Pull request merged**